docs/custom-flows/email-password.mdx

```diff
@@ -387,6 +387,146 @@ To sign up a user using their email and password, you must:
+
+      const [firstName, setFirstName] = React.useState("");
+      const [lastName, setLastName] = React.useState("");
+      const [emailAddress, setEmailAddress] = React.useState("");
+      const [password, setPassword] = React.useState("");
+      const [pendingVerification, setPendingVerification] = React.useState(false);
+      const [code, setCode] = React.useState("");
+
+      // Start the sign up process.
+      const onSignUpPress = async () => {
+        if (!isLoaded) {
+          return;
+        }
+
+        try {
+          await signUp.create({
+            firstName,
+            lastName,
+            emailAddress,
+            password,
+          });
+
+          // send the email.
+          await signUp.prepareEmailAddressVerification({ strategy: "email_code" });
+
+          // change the UI to our pending section.
+          setPendingVerification(true);
+        } catch (err: any) {
+          console.error(JSON.stringify(err, null, 2));
+        }
+      };
+
+      // Try to verify the user with the provided email code.
+      const onPressVerify = async () => {
+        if (!isLoaded) {
+          return;
+        }
+
+        try {
+          const completeSignUp = await signUp.attemptEmailAddressVerification({
+            code,
+          });
+
+          await setActive({ session: completeSignUp.createdSessionId });
+        } catch (err: any) {
+          console.error(JSON.stringify(err, null, 2));
+        }
+      };
+
+      return (
+        <View>
+          {!pendingVerification && (
+            <View>
+              <View>
+                <TextInput
+                  autoCapitalize="none"
+                  value={firstName}
+                  placeholder="First Name..."
+                  onChangeText={(firstName) => setFirstName(firstName)}
+                />
+              </View>
+              <View>
+                <TextInput
+                  autoCapitalize="none"
+                  value={lastName}
+                  placeholder="Last Name..."
+                  onChangeText={(lastName) => setLastName(lastName)}
+                />
+              </View>
+              <View>
+                <TextInput
+                  autoCapitalize="none"
+                  value={emailAddress}
+                  placeholder="Email..."
+                  onChangeText={(email) => setEmailAddress(email)}
+                />
+              </View>
+
+              <View>
+                <TextInput
+                  value={password}
+                  placeholder="Password..."
+                  placeholderTextColor="#000"
+                  secureTextEntry={true}
+                  onChangeText={(password) => setPassword(password)}
+                />
+              </View>
+
+              <TouchableOpacity onPress={onSignUpPress}>
+                <Text>Sign up</Text>
+              </TouchableOpacity>
+            </View>
+          )}
+          {pendingVerification && (
```

Comment:
Not necessary, but we have these in the next.js example so you can decide if we want these 🤷‍♀️ 

```suggestion
          // Display the verification form to capture the OTP code
          {pendingVerification && (
```